From 887ceda573d90ddd4c1eb1f5c746a91a669e071f Mon Sep 17 00:00:00 2001
From: Nicolai Buchwitz <nb@tipi-net.de>
Date: Tue, 28 Jul 2026 22:20:21 +0200
Subject: [PATCH] arm64: dts: rp1: Mux SPI chip-select pins as GPIO

The spi0 and spi8 CS groups mux their pins to the peripheral function
while the pins are actually driven via cs-gpios. Under strict pinmux the
overlapping function and GPIO claims are rejected and the SPI host fails
to probe. Mux them as GPIO to match how they are used.

Signed-off-by: Nicolai Buchwitz <nb@tipi-net.de>
---
 arch/arm64/boot/dts/broadcom/rp1.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm64/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm64/boot/dts/broadcom/rp1.dtsi
@@ -930,7 +930,7 @@
 			};
 
 			rp1_spi0_cs_gpio7: rp1_spi0_cs_gpio7 {
-				function = "spi0";
+				function = "gpio";
 				pins = "gpio7", "gpio8";
 				bias-pull-up;
 			};
@@ -984,7 +984,7 @@
 			};
 
 			rp1_spi8_cs_gpio52: rp1_spi8_cs_gpio52 {
-				function = "spi0";
+				function = "gpio";
 				pins = "gpio52", "gpio53";
 				bias-pull-up;
 			};
